src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/FlagVisitsTask.kt

```diff
@@ -65,50 +71,60 @@ class VisitTask(
       }
     }
 
-    log.debug("Finished flagVisits task.")
+    LOG.debug("Finished flagVisits task.")
   }
 
   private fun flagVisit(visit: VisitDto, noticeDays: Int, isRetry: Boolean = false): Boolean {
     var retry = false
-
-    log.debug("Started check, visit with reference - {}, prisoner id - {}, prison code - {}, start time - {}, end time - {}", visit.reference, visit.prisonerId, visit.prisonCode, visit.startTimestamp, visit.endTimestamp)
+    var reason: String? = null
     var sessions = emptyList<VisitSessionDto>()
 
-    var exception: Exception? = null
-    try {
-      sessions = sessionService.getVisitSessions(prisonCode = visit.prisonCode, prisonerId = visit.prisonerId, minOverride = noticeDays, maxOverride = noticeDays)
-    } catch (e: PrisonerNotInSuppliedPrisonException) {
-      exception = e
-    } catch (e: Exception) {
-      // only log this if the visit is being retried
-      if (isRetry) {
-        exception = e
-      } else {
-        retry = true
+    LOG.debug("Started check, visit with reference - {}, prisoner id - {}, prison code - {}, start time - {}, end time - {}", visit.reference, visit.prisonerId, visit.prisonCode, visit.startTimestamp, visit.endTimestamp)
+    val notifications = getVisitNotifications(visit.reference)
+    val hasNotifications = notifications.isNotEmpty()
+
+    if (hasNotifications) {
+      reason = notifications.joinToString(", ") { it.description }
+    } else {
+      try {
+        sessions = sessionService.getVisitSessions(prisonCode = visit.prisonCode, prisonerId = visit.prisonerId, minOverride = noticeDays, maxOverride = noticeDays)
+      } catch (e: PrisonerNotInSuppliedPrisonException) {
+        reason = "Prisoner - ${visit.prisonerId} has moved prison"
+        LOG.debug("Prisoner {} has moved prison", visit.prisonerId)
+      } catch (e: Exception) {
+        // only log this if the visit is being retried
+        LOG.info("Exception thrown when retrieving visit sessions for the following parameters - prison code - {}, prisonerId - {}, minOverride - {}, maxOverride - {}", visit.prisonCode, visit.prisonerId, noticeDays, noticeDays)
+        if (!isRetry) {
+          retry = true
+        }
```

Comment:
is this correct? E.g. isRetry = true, this if statement will not trigger? Not sure if it's just the name confusing me haha

        if (!isRetry) {
          retry = true
        }

Reply:
updated retry to markForRetry and added comments.